roles/generate_agent_iso/tasks/main.yml

```diff
@@ -5,12 +5,22 @@
     mode: 0755
     recurse: true
 
-- name: Generate ISO
-  ansible.builtin.shell:
-    cmd: "{{ agent_based_installer_path }} --log-level=debug agent create image"
-    chdir: "{{ manifests_dir }}"
-  environment:
-    XDG_RUNTIME_DIR: "{{ config_file_path }}"
+- name: Generate an ISO and print message if there is a failure
+  block:
+    - name: Generate ISO
+      ansible.builtin.shell:
+        cmd: "{{ agent_based_installer_path }} --log-level=debug agent create image"
+        chdir: "{{ manifests_dir }}"
+      environment:
+        XDG_RUNTIME_DIR: "{{ config_file_path }}"
+      register: gen_iso
+  rescue:
+    - name: Print message about nmstate package missing
+      debug:
+        msg: "nmstate package missing in the bastion. See Dependencies section in README.md for more details."
+      when:
+        - gen_iso.stdout is defined
+        - gen_iso.stdout | regex_search("nmstatectl(.*) executable file not found"
```

Comment:
It writes this out to stdout not stderr?

Reply:
good catch, fixed. I'll run a test that will fail in this task because a lack of nmstate pkg to validate the output